torchvision/models/detection/_utils.py

```diff
@@ -201,15 +202,24 @@ def decode_single(self, rel_codes, boxes):
         pred_w = torch.exp(dw) * widths[:, None]
         pred_h = torch.exp(dh) * heights[:, None]
 
-        pred_boxes = torch.zeros_like(rel_codes)
-        # x1
-        pred_boxes[:, 0::4] = pred_ctr_x - 0.5 * pred_w
-        # y1
-        pred_boxes[:, 1::4] = pred_ctr_y - 0.5 * pred_h
-        # x2
-        pred_boxes[:, 2::4] = pred_ctr_x + 0.5 * pred_w
-        # y2
-        pred_boxes[:, 3::4] = pred_ctr_y + 0.5 * pred_h
+        # the following lines involve advanced indexing that is not yet supported in ONNX tracing
+        if torchvision._is_tracing():
+            pred_boxes1 = pred_ctr_x - torch.tensor(0.5, dtype=pred_ctr_x.dtype) * pred_w
```

Comment:
I would be in favor of replacing the old implementation with this new one. It can be entirely fused by the JIT compiler I believe (which would make things faster, if we use `cat` instead of `stack`, but I can change it later), and I don't think it will be that much slower maybe.
Can you just do a quick speed test to see how much it would impact performance to replace one by the other? Can be a simple benchmark of this particular function, with representative sizes.

Reply:
I did a quick test on these couple of lines with random inputs of the same shape.
The old implementation runs in around 2.8 millisec, and the new one in 19 millisec (same result using cat or stack)..

Should I keep it as is, or still replace the old implementation with this new one?